pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -305,7 +305,13 @@ self: super: builtins.intersectAttrs super {
   ghc-debug-brick  = enableSeparateBinOutput super.ghc-debug-brick;
   nixfmt  = enableSeparateBinOutput super.nixfmt;
   calligraphy = enableSeparateBinOutput super.calligraphy;
-  niv = enableSeparateBinOutput (self.generateOptparseApplicativeCompletions [ "niv" ] super.niv);
+  niv = overrideCabal (drv: {
```

Comment:
With three nested function calls I think this would maybe be more readable using `lib.pipe`.

Reply:
No need to change this now, just something I noticed.

Reply:
Yeah this was my first encounter with the haskell packageset, and I found it .. unusual to say the least. So found it hard to determine what would be considered "neat".